troposphere/serverless.py

```diff
@@ -31,6 +31,19 @@ def policy_validator(x):
                          + " policy documents")
 
 
+class DeadLetterQueue(AWSProperty):
+    props = {
+        'Type': (basestring, False),
+        'TargetArn': (basestring, False)
+    }
+
+    def validate(self):
+        valid_types = ['SQS', 'SNS']
+        if self.properties['Type'] not in valid_types:
+            raise ValueError('Type must be either SQS or SNS')
+        return self.properties['Type']
```

Comment:
Property validators do return a type but object validators (like this one) don't return anything.